servers/audio/effects/eq.cpp

```diff
@@ -103,10 +106,12 @@ void EQ::recalculate_band_coefficients() {
 
 		//printf("band %i, precoefs = %f,%f,%f\n",i,c2a,c2b,c2c);
 
-		double r1, r2; //roots
+		// roots
+		double r1 = std::numeric_limits<double>::quiet_NaN();
```

Comment:
Since using them uninitialized will never happen, I think we don't need to use this kind of approach, just initialize them to 0 to satisfy the compiler.

Reply:
Alsop, is using `std::` allowed here?

Reply:
Not so much right now, but this will change with 4.0 when we start using C++14 and some `std` components. I'm not sure yet if `std::numeric_limits` would be part of the "now allowed" modules, we'll see.

Reply:
My initial thought was exactly the same: just initialise them to 0. However, I ended up here by make the following assumptions:

1. We shouldn't be writing code that's dependent on the implementation of the functions being used.
2. Complex roots are valid roots; so a different implementation may return 2 when (b * b - 4 * a * c) < 0.
3. As suggested by the test that highlighted this issue, the philosophy is to treat warnings as errors.
4.  If the compiler isn't able to determine whether or not the variables will be used uninitialised, I think we should accept it and not think we're better than it. There are many examples on [StackOverflow](https://stackoverflow.com/search?q=may+be+used+uninitialized) highlighting how easy and often this mistake is made.

Then, if we initialise them to zero, there is no way to check whether or not they have been initialised after the initialisation function, because 0 is a valid value too. It's not the same as, for example, setting a pointer to 0 and then checking that it's initialised by checking that it's not zero.

In the general case, assigning an arbitrary default value is no better than turning the warning off. In fact, it's worse, because you can't turn it back on again without knowing where in the code it was turned off.

However, if we can't use `std::numeric_limits`, I don't know what else to suggest.

Reply:
> Complex roots are valid roots

`double` does not support complex numbers unless you use `std::complex<double>`.

Reply:
I've updated the commit to use 0.